migrator-tool/src/main/scala/za/co/absa/spline/migrator/BatchMigratorActor.scala

```diff
@@ -56,23 +56,26 @@ class MigratorActor(conf: MigratorConfig)
 
     def processPage(page: PageRequest, prevTotals: Stats): Unit = {
       context become processingPage(Stats.emptyTree.copy(parentStats = prevTotals), None)
-      mongoActor ! MongoActor.GetLineages(page)
+      spline03Actor ! Spline03Actor.GetExistingLineages(page)
 
       def processingPage(pageStats: TreeStats, pageActualSize: Option[Int]): Receive = {
-        case lineage: DataLineage =>
-          arangoActor ! ArangoActor.LineagePersistRequest(lineage)
-
         case PageSize(pageSize) =>
           onCountsUpdate(pageStats, Some(pageSize)) ensuring pageActualSize.isEmpty
 
-        case resp: ArangoActor.LineagePersistResponse =>
-          resp.result match {
-            case Success(_) =>
-              onCountsUpdate(pageStats.incSuccess, pageActualSize)
-            case Failure(e: Throwable) =>
-              onCountsUpdate(pageStats.incFailure, pageActualSize)
-              log.error(e, e.getMessage)
-          }
+        case DataLineageLoaded(lineage: DataLineage) =>
+          val (executionPlan, maybeExecutionEvent) = new DataLineageToExecPlanWithEventConverter(lineage).convert()
+          spline04Actor ! Save(executionPlan, maybeExecutionEvent)
+
+        case SaveSuccess(_) =>
+          onCountsUpdate(pageStats.incSuccess, pageActualSize)
+
+        case DataLineageLoadFailure(_, e) =>
```

Comment:
Unfortunately this wouldn't work because of ambiguous `e` variable mapping. But since `e` is only used for logging, I think it will improve in #269 as the logging will most likely go away and will be replaced by some kind of recovering code.